ably/rest/channel.py

```diff
@@ -149,18 +149,33 @@ def encrypted(self):
     def options(self):
         return self.__options
 
+    @options.setter
+    def options(self, options):
+        self.__options = options
+
+        if options and options.encrypted:
+            self.__cipher = get_cipher(options.cipher_params)
+        else:
+            self.__cipher = None
+
 
 class Channels(object):
     def __init__(self, rest):
         self.__ably = rest
-        self.__attached = {}
+        self.__attached = OrderedDict()
 
     def get(self, name, options=None):
         if isinstance(name, six.binary_type):
             name = name.decode('ascii')
+
```

Comment:
I will merge now, but where we write code to add 2/3 support, it may be nice to add a small comment with an explanation for future us.